stor/obs.py

```diff
@@ -74,6 +75,14 @@ def __init__(self, pth):
     copy = utils.copy
     copytree = utils.copytree
 
+    def dirname(self):
+        """
+        Return the directory name of self.
+        """
+        assert self.startswith(self.drive)
+        rest = self[len(self.drive):]
+        return type(self)(self.drive + os.path.dirname(rest))
```

Comment:
Done. Previous version was working as well, but I think this one is more explicit.